src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java

```diff
@@ -757,15 +757,14 @@ protected final JsonNode _fromFloat(JsonParser p, DeserializationContext ctxt,
         if (ctxt.isEnabled(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)) {
             // [databind#4194] Add an option to fail coercing NaN to BigDecimal
             // Currently, Jackson 2.x allows such coercion, but Jackson 3.x will not
-            if (p.isNaN() && ctxt.isEnabled(JsonNodeFeature.FAIL_ON_NAN_TO_BIG_DECIMAL_COERCION)) {
-                ctxt.handleWeirdNumberValue(handledType(), p.getDoubleValue(),
-                        "Cannot convert NaN into BigDecimal");
-            }
-            try {
-                return _fromBigDecimal(ctxt, nodeFactory, p.getDecimalValue());
-            } catch (NumberFormatException nfe) {
-                // fall through - BigDecimal does not support values like NaN
+            if (p.isNaN()) {
+                if (ctxt.isEnabled(JsonNodeFeature.FAIL_ON_NAN_TO_BIG_DECIMAL_COERCION)) {
+                    return (JsonNode) ctxt.handleWeirdNumberValue(handledType(), p.getDoubleValue(),
```

Comment:
is `return (JsonNode)` part added because of warning? `ctxt.handleWeirdNumberValue` throws an error

Reply:
It is because it is possible to specify a `DeserializationProblemHandler` which can "fix" the problem and return actual valid `JsonNode` value :)
(hence, "handle" - prefix)

Usually an exception is thrown but it can be prevented by custom handler this way.

Reply:
Ahh I see. Thank you for explanation! 🥹